### ISSUE TITLE
Invalid cloud-init when using custom cloud-init config

### ISSUE THREAD

**Issue by nnoethling** [nutanix/docker-machine#107]
### Rancher Version
2.7.10

### Rancher Node Driver Version
3.6.0

### Expected Behavior
When using custom cloud-init config, the generated cloud-init config should be valid

### Actual Behavior
cloud-init config on node is invalid due to [deprecation of ssh-authorized-keys](https://cloudinit.readthedocs.io/en/latest/reference/modules.html#users-and-groups)
Seems like the driver adds the deprecated ssh-authorized-keys to the cloud-init config of the root user [here](https://github.com/nutanix/docker-machine/blob/71b521a20b54f5999e7a5c693cf9467174c1d373/machine/driver/driver.go#L400).
Deployment of the Nodes is working and Nodes become active in Cluster and can be used.
If you run syntax check for cloudinit `sudo cloud-init schema --system` the cloud-init is invalid.

### Details
If you run `sudo cloud-init schema --system` the command return errors because the config parameter `ssh-authorized-keys` is deprecated since version 18.3
Cloud-init seems to run anyway and executes everything.
But this could change in the future when cloud-init is removing the config parameter.

### Debug output
When using custom cloud-init config the generated cloud-init config is invalid.
```
# sudo cloud-init schema --system
Invalid cloud-config /var/lib/cloud/instances/20b6b7a3-9e5d-400c-af2d-46908cd6852a/cloud-config.txt
Error: Cloud config schema errors:
users.1: {'name': 'root', 'ssh-authorized-keys': ['ssh-rsa AAA...Hrno'], 'sudo': 'ALL=(ALL) NOPASSWD:ALL'} is not valid under any of the given schemas
```

## Steps to reproduce:

1. Enable Nutanix node driver in Rancher with version 3.6.0
2. Deploy new RKE2 Cluster using Nutanix node driver
3. Create custom cloud-init config and add users to it
4. Fill the cloud-init field in the cluster node driver settings
5. Deploy the nodes/cluster
6. Login to one of the created Nodes and run `sudo cloud-init schema --system`

**Comment by nnoethling** [comment]
OS: ubuntu 22.04

**Comment by tuxtof** [comment]
there is no issue with ubuntu 22.04 on my side
- can you share the custom cloud-init config you try to use
- is it with RKE1 or RKE2 ?
- how do you trigger the cluster creation , UI , API , CRD ?

**Comment by nnoethling** [comment]
> can you share the custom cloud-init config you try to use

```
#cloud-config 
ca_certs:
  trusted:
    - |
      -----BEGIN CERTIFICATE-----     
      ...       
      -----END CERTIFICATE-----
manage_etc_hosts: true
ntp:
  enabled: true
  servers:
    - 1.2.3.4
    - 4.3.2.1
users:
  - groups: sudo
    name: ubuntu
    ssh-authorized-keys:
      - ssh-rsa AAAAC1yc2...43O
    sudo:
      - ALL=(ALL) NOPASSWD:ALL
runcmd:
  - apt update
  - apt -y install nfs-common
```

> is it with RKE1 or RKE2 ?

RKE2 clusters

> how do you trigger the cluster creation , UI , API , CRD ?

We are using UI

**Comment by nnoethling** [comment]
Fixed by #108
